packages/devtools_app/lib/src/charts/treemap.dart

```diff
@@ -0,0 +1,517 @@
+import 'package:flutter/material.dart';
+
+import '../trees.dart';
+import '../utils.dart';
+
+class Treemap extends StatefulWidget {
+  const Treemap({
+    @required this.rootNode,
+    @required this.levelsVisible,
+    @required this.width,
+    @required this.height,
+    @required this.onTap,
+  });
+
+  final TreemapNode rootNode;
+
+  /// The depth of children visible from this Treemap widget.
+  ///
+  /// For example, levelsVisible = 2 at root Treemap:
+  /// ```
+  /// _______________
+  /// |     Root    |
+  /// ---------------
+  /// |    l = 1    |
+  /// |  ---------  |
+  /// |  | l = 2 |  |
+  /// |  |       |  |
+  /// |  |       |  |
+  /// |  ---------  |
+  /// ---------------
+  /// ```
+  final int levelsVisible;
+
+  final double width;
+
+  final double height;
+
+  final VoidCallback onTap;
+
+  @override
+  _TreemapState createState() => _TreemapState();
+}
+
+enum PivotType { pivotByMiddle, pivotBySize }
+
+class _TreemapState extends State<Treemap> {
+  static const double minHeightToDisplayText = 20.0;
+  PivotType pivotType = PivotType.pivotBySize;
+
+  TreemapNode rootNode;
+
+  @override
+  void initState() {
+    super.initState();
+    rootNode = widget.rootNode;
+  }
+
+  bool get shouldDisplayText => widget.height > minHeightToDisplayText;
+
+  void cellOnTap(TreemapNode child) {
+    if (child != null) {
+      setState(() {
+        rootNode = child;
+      });
+    }
+  }
+
+  /// Computes the total size of a given list of treemap nodes.
+  int computeByteSizeForNodes({
+    @required List<TreemapNode> nodes,
+    @required int startIndex,
+    @required int endIndex,
```

Comment:
on second thought, let's make these optional named parameters that default to 0 and nodes.length - 1 respectively.

You won't be able to set `endIndex` to a default of `node.length - 1` in the params, but you can write `endIndex ??= nodes.length - 1;` at the beginning of the method.

Reply:
Done